integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedRDDUtils.scala

```diff
@@ -47,10 +46,24 @@ object DistributedRDDUtils {
                                               .getDistributable.getSegment.getIndexSize)
     val executorCache = DistributedRDDUtils.executorToCacheSizeMapping
     // check if any executor is dead.
-    val invalidExecutors = executorCache.keySet().asScala.diff(executorsList)
-    if (invalidExecutors.nonEmpty) {
+    val invalidHosts = executorCache.keySet().asScala.diff(executorsList.keySet)
```

Comment:
There can be a chance of executor down also, not just complete machine down.  It is better to get the invalid executors.